sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionAnsiErrorsSuite.scala

```diff
@@ -322,4 +329,71 @@ class QueryExecutionAnsiErrorsSuite extends QueryTest
         "columnName" -> "`col`")
     )
   }
+
+  test("user-facing runtime errors") {
```

Comment:
nit (if you need to push some changes)

```suggestion
  test("SPARK-46922: user-facing runtime errors") {
```